jormungandr/src/topology/quarantine.rs

```diff
@@ -62,24 +68,30 @@ impl ReportRecords {
     }
 
     /// Returns whether the node has been quarantined or not.
-    pub fn report_node(&mut self, topology: &mut poldercast::Topology, node: Peer) -> bool {
+    pub fn report_node(
+        &mut self,
+        topology: &mut poldercast::Topology,
+        node: Peer,
+    ) -> ReportNodeResult {
         if self.report_whitelist.contains(&node.address()) {
             tracing::debug!(
                 node = %node.address(),
                 id=?node.id(),
                 "quarantine whitelists prevents this node from being reported",
             );
-            false
+            ReportNodeResult::Ok
         } else if self.report_grace.contains(&node.id()) {
             tracing::trace!(node = %node.address(), id=?node.id(), "not reporting node in grace list");
-            false
+            ReportNodeResult::Ok
         } else {
             let mut peer_info = PeerInfo::from(node);
             tracing::debug!(node = %peer_info.address, id=?peer_info.id, ?self.report_duration, "reporting node");
             // If we'll handle report reasons other that a connectivity issue in the future, we may want to
             // demote a peer all the way down to dirty in case of a serious violation.
             topology.remove_peer(peer_info.id.as_ref());
 
+            let mut result = ReportNodeResult::Trusted;
```

Comment:
Not too clear what `ReportNodeResult::Trusted` should mean. At this stage a node is either in the `pool` pool, which is standard trust, or quarantined (where is correctly flagged as `ReportNodeResult::Quarantine`), but it cannot be in the `trusted` state.

Reply:
The thing is that we do not really put trusted nodes in quarantine (see the code below), it is sort of a special status. And these nodes should not be counted in the quarantine counter.

Reply:
But once they're reported they're no longer trusted. The logic seems good, is just that I find the names a little confusing, like `ReportNodeResult::Quarantine` is the status *after* the report, but `ReportNodeResult::Trusted` is *before*.

Reply:
Maybe, `Preferred` would be the right term?

Reply:
mmm, something like `LightReport`? The peer is indeed reported and removed from the topology, but it's not quarantined. We do not have a word for that, but  `LightReport` should convey the idea I hope